### PR TITLE
Feature/refactor document popup related code #29

### PR DESCRIPTION
**Pull request by 4lm**
Hi @nesnoj this is the pull request to issue #29. Please have a look and feel free to merge if you approve the changes. You find more details concerning the changes in: https://github.com/rl-institut/WAM_APP_stemp_abw/issues/29#issuecomment-485869462

### CODE REVIEW

File: urls.py [hunk 1]
```diff
@@ -45,70 +43,14 @@
...
-        )
-    ]
+    path('popupjs/{}/<int:pk>/'.format(name), dview.as_view(
+        template_name='stemp_abw/popups/js_popup.html'),
+         name='{}-js'.format(name))
+    for name, dview in detail_views.items()
 )
 
-# search JSON data views classes and append to URLs
+# Search JSON data-view-classes and append to URLs
```
Comment (nesnoj) [inline urls.py]: Notice: In composite nouns in English, you only use hyphens if parts of speech are different (e.g. noun+adverb), but not with 2 or more nouns

File: views/detail_views.py [hunk 2]
```diff
@@ -1,52 +1,87 @@
...
+        in the metadata config file then the returned list is empty.
+
+        Parameters
+        ----------
+        metadata : :obj:`ConfigObj`
+        app_name : :obj:`str`
+
+        Returns
+        -------
+        :obj:`list` of :obj:`Source`
```
Comment (nesnoj) [inline views/detail_views.py]: References to non-builtin classes should be done using
```
:class:`<path>`
```
in this case
```
:class:`wam.meta.models.Source`
```

File: urls.py [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
 
-from . import views
-from inspect import getmembers, isclass
 from meta.models import Source
 from meta.views import AppListView, AssumptionsView
-#from stemp_abw.views.serial_views import SplitDataView
+from stemp_abw.app_settings import MAP_DATA_CACHE_TIMEOUT
+from . import views
 
 app_name = 'stemp_abw'
```
Comment (nesnoj) [inline urls.py:12]: Shouldn't we set this in app_settings.py?
`APP_NAME = 'stemp_abw'`
and import where needed..

File: views/detail_views.py [hunk 4]
```diff
@@ -1,52 +1,87 @@
...
+                    pass
+
+    def get_context_data(self, **kwargs):
+        context = super(MasterDetailView, self).get_context_data(**kwargs)
+
+        context['title'] = LABELS['layers'][self.model.name]['title']
+        context['text'] = LABELS['layers'][self.model.name]['text']
+
+        # Get app_name from request
+        app_name = self.request.resolver_match.app_name
```
Comment (nesnoj) [inline views/detail_views.py:58]: Why not using the `APP_NAME` from app_settings.py here? (see above)

**Review by nesnoj — CHANGES_REQUESTED** [review]
Good work, thank you!
Find some comments in the code

File: urls.py (hunk 1 above)
Comment (4lm) [reply urls.py]: Yes, there is one hyphen to much, I learned that you use hypens for compound nouns (except for the last word) if you have 3+ words. On the Internet I found concerning hyphens and nouns: ["Rule 2b. When writing out new, original, or unusual compound nouns, writers should hyphenate whenever doing so avoids confusion"](https://www.grammarbook.com/punctuation/hyphens.asp). So I will change this to "data-view classes".

File: views/detail_views.py (hunk 2 above)
Comment (4lm) [reply views/detail_views.py]: Thanks for the info, as spoken I will use this solution but change `:class:` to `:obj:`.

File: urls.py (hunk 3 above)
Comment (4lm) [reply urls.py:12]: As spoken app_name is created by Django. Maybe we find a better solution in the future. But wil leave this solution for right now.

File: views/detail_views.py (hunk 4 above)
Comment (4lm) [reply views/detail_views.py:58]: See answer above.

**Comment by 4lm** [comment]
I added the changes from the review and spoke with @nesnoj off-line. It's OK to directly merge the pull request after the changes of the review process 16ca8cc are commited. So I will do this now :)

**Pull request merged**